src/csvcubed/models/cube/qb/validationerrors.py

```diff
@@ -324,3 +326,136 @@ def __post_init__(self):
+    """
+
+    column_names: List[str]
+    column_names_concatenated: str = field(init=False)
+
+    @classmethod
+    def get_error_url(cls) -> str:
+        return "http://purl.org/csv-cubed/err/dup-measure"
+
+    def __post_init__(self):
+        self.column_names_concatenated = ", ".join(self.column_names)
+        self.message = f"In the pivoted shape, two or more observation value columns cannot be represented by identical measures. {self.column_names_concatenated}"
+
+
+@dataclass
+class AttributeNotLinkedError(SpecificValidationError):
+    """
+    An error to inform the user that the units or attribute column is defined but it is not linked to the obs column
+    """
+
+    attribute_column_title: str
+
+    @classmethod
+    def get_error_url(cls) -> str:
+        return "http://purl.org/csv-cubed/err/att-not-linked"
+
+    def __post_init__(self):
+        self.message = f"Units or attribute column '{self.attribute_column_title}' is defined but it is not linked to an observed values column"
+
+
+@dataclass
+class LinkedObsColumnDoesntExistError(SpecificValidationError):
+    """
+    An error to infrom the user that the units or attribute column is defined for which obs val column doesn't appear to exist.
+    """
+
+    alleged_obs_val_column_title: str
+    attribute_column_title: str
+
+    @classmethod
+    def get_error_url(cls) -> str:
+        return "http://purl.org/csv-cubed/err/link-obs-col-not-exist"
+
+    def __post_init__(self):
+        self.message = f"The unit or attribute column '{self.attribute_column_title}' is defined for an observed value column '{self.alleged_obs_val_column_title}' that doesn't appear to exist."
+
+
+@dataclass
+class LinkedToNonObsColumnError(SpecificValidationError):
+    """
+    An error to infrom the user that units or attribute column is defined in which
+    the linked obs val column isn't actually an observations column.
+    """
+
+    alleged_obs_val_column_title: str
+    attribute_column_title: str
+
+    @classmethod
+    def get_error_url(cls) -> str:
+        return "http://purl.org/csv-cubed/err/link-non-obs-col"
+
+    def __post_init__(self):
+        self.message = f"Units or attribute column '{self.attribute_column_title}' is defined but the linked observation column '{self.alleged_obs_val_column_title}' is not actually an observation column"
+
+
+@dataclass
+class HybridShapeError(SpecificValidationError):
+    """
+    An Error where there are mutliple obs val columns defined without measures, and at least one measure column defined.
+    This is an erroneous hybrid between standard and pivoted shape.
+    """
+
+    not_linked_obs_val_cols: List[QbColumn[QbObservationValue]]
+    measure_cols: List[QbColumn[QbMultiMeasureDimension]]
+
+    @classmethod
+    def get_error_url(cls) -> str:
+        return "http://purl.org/csv-cubed/err/hybrid-shape"
+
+    def __post_init__(self):
+
+        not_linked_cols = ", ".join(str(self.not_linked_obs_val_cols))
```

Comment:
If you are going to pass `QbColumn`s into here, then you shouldn't be directly converting them to strings like this for the message, you should be extracting their `csv_column_title` since that's what the users will understand.